src/Pages/Detail/Detail.js

```diff
@@ -1,7 +1,175 @@
 import React, { Component } from 'react';
+import CategoryImage from './CategoryImage/CategoryImage';
+import AnotherItem from './AnotherItem/AnotherItem';
+import SimilarItem from './SimilarItem/SimilarItem';
+import WantEatContainer from './WantEatContainer/WantEatContainer';
+import EvaluationContainer from './EvaluationContainer/EvaluationContainer';
+import './Detail.scss';
 
 export default class Detail extends Component {
+  constructor(props) {
+    super(props);
+    this.sliderRef = React.createRef();
+    this.itemRef = React.createRef();
+    this.state = {
+      transFormPosition: 0,
+      eachdatalist: [],
+      anotherItemLocation: null,
+    };
+  }
+
+  // callApi = () => {
+  //   fetch('http://10.58.2.238:8000/products?category=전체')
+  //     .then(res => res.json())
+  //     .then(data =>
+  //       this.setState({
+  //         eachdatalist: data,
+  //       })
+  //     );
+  // };
+
+  // componentDidMount = () => {
+  //   this.callApi();
+  // };
+
+  componentDidMount = () => {
+    const sliderElement = this.sliderRef.current;
+    const itemElement = this.itemRef.current;
+    const itemWidth = Math.ceil(itemElement.getBoundingClientRect().width);
```

Comment:
- itemElement 변수를 거치는 것 보다, 그냥 itemWidth 를 ref 를 사용해서 바로 계산하는게 더 좋아보입니다!
```suggestion
    const itemWidth = Math.ceil(this.itemRef.current.getBoundingClientRect().width);
```